### PR TITLE
Add worker package comment

### PR DESCRIPTION
**Pull request by xtreme-sameer-vohra**
# Why do we need this PR?
Add comment to package top level file to clarify the purpose of a package.
Inspired by @cirocosta's suggestion & goland docs [strings](https://golang.org/src/strings/strings.go?s=2326:2362#L5) 

# Changes proposed in this pull request
Add comment to atc/worker package to clarify its purpose

# Reviewer Checklist
> This section is intended for the core maintainers only, to track review progress. Please do not
> fill out this section.
- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the [BOSH](https://github.com/concourse/concourse-bosh-release) 
      and [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for the [integration tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env) (for example, if they are Garden configs that are not displayed in the `--help` text).

### CODE REVIEW

File: atc/worker/worker.go [hunk 1]
```diff
@@ -1,5 +1,9 @@
 package worker
 
+// This package will eventually evolve to becoming a concrete implementation of a runtime
```
[the remark this answers is not in the document]
Comment (xtreme-sameer-vohra) [reply atc/worker/worker.go]: Thx, fixed the docs issue.

Yes, totally. We can incrementally add more context to cover the interfaces. The `Client` is the main one that I have called out for now.

File: atc/worker/worker.go [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+// As such, Concourse core shouldn't depend on abstractions defined in this package or its child packages
+// General Runtime abstractions will be ported over to the Runtime package
+//   The Client interface is the main interface that is consumed by Concourse core that will be shifted to the Runtime package
```
Comment (cirocosta) [inline atc/worker/worker.go]:
```suggestion
// The Client interface is the main interface that is consumed by Concourse core that will be shifted to the Runtime package
```

File: atc/worker/worker.go [hunk 3]
```diff
@@ -1,3 +1,7 @@
+// This package will eventually evolve to becoming a concrete implementation of a runtime
```
Comment (cirocosta) [inline atc/worker/worker.go]:
```suggestion
// Package worker will eventually evolve to becoming a concrete implementation of a runtime
```

package docs are meant to start with `Package <pkg_name>`, e.g.:

```
// Package io provides basic interfaces to I/O primitives.
// Its primary job is to wrap existing implementations of such primitives,
// such as those in package os, into shared public interfaces that
// abstract the functionality, plus some other related primitives.
```

https://github.com/golang/go/blob/71239b4f491698397149868c88d2c851de2cd49b/src/io/io.go#L5

**Comment by xtreme-sameer-vohra** [comment]
@cirocosta 
thx for the suggestions. accepted the suggestions and squashed the commits for a cleaner history.

**Pull request merged**